packages/aws-cdk-lib/aws-s3/lib/bucket.ts

```diff
@@ -2085,6 +2085,11 @@ export class Bucket extends BucketBase {
 
     function parseLifecycleRule(rule: LifecycleRule): CfnBucket.RuleProperty {
       const enabled = rule.enabled ?? true;
+      if ((rule.expiredObjectDeleteMarker != null && rule.expiredObjectDeleteMarker != undefined)
+      && (rule.expiration || rule.expirationDate || self.parseTagFilters(rule.tagFilters))) {
+        // ExpiredObjectDeleteMarker cannot be specified with ExpirationInDays, ExpirationDate, or TagFilters.
+        throw new Error('ExpiredObjectDeleteMarker cannot be specified with expiration, ExpirationDate, or TagFilters.');
+      }
```

Comment:
```suggestion
      if ((rule.expiredObjectDeleteMarker)
      && (rule.expiration || rule.expirationDate || this.parseTagFilters(rule.tagFilters))) {
        // ExpiredObjectDeleteMarker cannot be specified with ExpirationInDays, ExpirationDate, or TagFilters.
        throw new Error('ExpiredObjectDeleteMarker cannot be specified with expiration, ExpirationDate, or TagFilters.');
      }
```

Reply:
The reason I do not use the "if ((rule.expiredObjectDeleteMarker))" is the type of "expiredObjectDeleteMarker" is boolean. If its value is false, the validation here will not work.  I assume "false" could also not be specified with ExpirationInDays, ExpirationDate, or TagFilter according to public doc.

See if it makes sense.

Reply:
Btw, the "this.parseTagFilters(rule.tagFilters)" will not compile, and there is same "this.parseTagFilters(rule.tagFilters)" usage at line 2112/2117. My change is to stay consistent with existing code.

See if it makes sense.